pages/studentPortal/labs/multiplication-connect/Index.tsx

```diff
@@ -35,69 +25,121 @@ export const calculateWinner = (grid: GameBoardBlock[]) => {
     grid.filter((i) => i.id >= 25 && i.id < 30),
     grid.filter((i) => i.id >= 30 && i.id < 35),
   ];
-
+  let player: SelectedBy;
+  isPlayerOne
+    ? (player = SelectedBy.PlayerOne)
+    : (player = SelectedBy.PlayerTwo);
   for (let i = 0; i < rows.length; i++) {
     // rows.length == board height == 7
     // rows[i].length == board width == 5
-
     for (let index = 0; index < rows[i].length - 3; index++) {
       // Horizontal check
-      rows[i][index].isSelected &&
-      rows[i][index + 1].isSelected &&
-      rows[i][index + 2].isSelected &&
-      rows[i][index + 3].isSelected
-        ? console.log("(horizontal) Four in a row!")
+      rows[i][index].selectedBy == player &&
+      rows[i][index + 1].selectedBy == player &&
+      rows[i][index + 2].selectedBy == player &&
+      rows[i][index + 3].selectedBy == player
+        ? console.log(player, "(horizontal) Four in a row!")
         : "";
     }
-
     if (i < rows.length - 3) {
       // Vertical check
       for (let index = 0; index < rows[i].length; index++) {
-        rows[i][index].isSelected &&
-        rows[i + 1][index].isSelected &&
-        rows[i + 2][index].isSelected &&
-        rows[i + 3][index].isSelected
-          ? console.log("(vertical) Four in a row!")
+        rows[i][index].selectedBy == player &&
+        rows[i + 1][index].selectedBy == player &&
+        rows[i + 2][index].selectedBy == player &&
+        rows[i + 3][index].selectedBy == player
+          ? console.log(player, "(vertical) Four in a row!")
           : "";
       }
     }
-
     if (i >= 3) {
       // Ascending diagonal check
       for (let index = 0; index < rows[i].length - 3; index++) {
-        rows[i][index].isSelected &&
-        rows[i - 1][index + 1].isSelected &&
-        rows[i - 2][index + 2].isSelected &&
-        rows[i - 3][index + 3].isSelected
-          ? console.log("(ascending diagonal) Four in a row!")
+        rows[i][index].selectedBy == player &&
+        rows[i - 1][index + 1].selectedBy == player &&
+        rows[i - 2][index + 2].selectedBy == player &&
+        rows[i - 3][index + 3].selectedBy == player
+          ? console.log(player, "(ascending diagonal) Four in a row!")
           : "";
       }
-
       // Descending diagonal check
       for (let index = 3; index < rows[i].length; index++) {
-        rows[i][index].isSelected &&
-        rows[i - 1][index - 1].isSelected &&
-        rows[i - 2][index - 2].isSelected &&
-        rows[i - 3][index - 3].isSelected
-          ? console.log("(descending diagonal) Four in a row!")
+        rows[i][index].selectedBy == player &&
+        rows[i - 1][index - 1].selectedBy == player &&
+        rows[i - 2][index - 2].selectedBy == player &&
+        rows[i - 3][index - 3].selectedBy == player
+          ? console.log(player, "(descending diagonal) Four in a row!")
```

Comment:
You should return the string here instead of console logging. Then you can test the output of this function

Reply:
Yeah for sure, still unfinished. Just logging to the console for the time being. I will implement the string return and a test cases in a PR soon.